java/client/src/org/openqa/selenium/firefox/internal/NewProfileExtensionConnection.java

```diff
@@ -154,7 +154,7 @@ protected void addWebDriverExtensionIfNeeded() {
 
   private static Optional<Extension> loadCustomExtension() {
     String xpiProperty = System.getProperty(FirefoxDriver.SystemProperty.DRIVER_XPI_PROPERTY);
-    if (xpiProperty != null) {
+    if (xpiProperty != null && !xpiProperty.equals("")) {
```

Comment:
instead of empty string, we probably should check that the File exists (next line down). Then log a warning if it doesn't and continue otherwise.